src/pipeline.ts

```diff
@@ -91,4 +91,8 @@ export class Pipeline<P extends Object = {}> implements PromiseLike<P> {
   ): Promise<TResult1 | TResult2> {
     return this.current.then(onfulfilled, onrejected);
   }
-}
+
+  toFactory(): Async.Factory<P> {
+    return Async.makeFactory(this.current as Promise<Async.Builder<P, keyof P>>);
```

Comment:
I don't love that we have to cast here, but I can't find anything more elegant.  `this.current` _is_ a valid `Async.Builder<P>`, but unfortunately makeFactory's signature won't quite allow it.  Maybe make a comment here explaining why the cast is valid?  Also you can just say `Promise<Async.Builder<P>>` (the second generic ar defaults to `keyof P`)

Reply:
this.current is Promise object, so I cannot help to use cast as Promise<Async.Builder<P, keyof P>>.
and I changed type parameter, thanks for review.